snowflake/src/main/java/org/apache/iceberg/snowflake/JdbcSnowflakeClient.java

```diff
@@ -136,60 +136,36 @@ void setQueryHarness(QueryHarness queryHarness) {
     this.queryHarness = queryHarness;
   }
 
-  /**
-   * For rare cases where PreparedStatements aren't supported for user-supplied identifiers intended
-   * for use in special LIKE clauses, we can sanitize by "broadening" the identifier with
-   * single-character wildcards and manually post-filter client-side.
-   *
-   * <p>Note: This sanitization approach intentionally "broadens" the scope of matching results;
-   * callers must be able to handle this method returning an all-wildcard expression; i.e. the
-   * caller must treat the usage of the LIKE clause as only an optional optimization, and should
-   * post-filter for correctness as if the LIKE clause wasn't present in the query at all.
-   */
-  @VisibleForTesting
-  String sanitizeIdentifierWithWildcardForLikeClause(String identifier) {
-    // Restrict identifiers to the "Unquoted object identifiers" synax documented at
-    // https://docs.snowflake.com/en/sql-reference/identifiers-syntax.html
-    //
-    // Use a strict allowlist of characters, replace everything *not* matching the character set
-    // with "_", which is used as a single-character wildcard in Snowflake.
-    String sanitized = identifier.replaceAll("[^a-zA-Z0-9_$]", "_");
-    if (sanitized.startsWith("$")) {
-      sanitized = "_" + sanitized.substring(1);
-    }
-    return sanitized;
-  }
-
   @Override
   public boolean databaseExists(SnowflakeIdentifier database) {
     Preconditions.checkArgument(
         database.type() == SnowflakeIdentifier.Type.DATABASE,
         "databaseExists requires a DATABASE identifier, got '%s'",
         database);
 
-    // Due to current limitations in PreparedStatement parameters for the LIKE clause in
-    // SHOW DATABASES queries, we'll use a fairly limited allowlist for identifier characters,
-    // using wildcards for non-allowed characters, and post-filter for matching.
-    final String finalQuery =
-        String.format(
-            "SHOW DATABASES LIKE '%s' IN ACCOUNT",
-            sanitizeIdentifierWithWildcardForLikeClause(database.databaseName()));
-    List<SnowflakeIdentifier> databases;
+    final String finalQuery = "SHOW SCHEMAS IN IDENTIFIER(?) LIMIT 1";
```

Comment:
We should ideally always specify the expected type of the identifier whenever there's a single expected type:

`SHOW SCHEMAS IN DATABASE IDENTIFIER(?) LIMIT 1`

That also helps us avoid accidentally querying the wrong object type if special characters are escaped incorrectly